Core/Src/bms.c

```diff
@@ -669,9 +671,47 @@ void updateGopherCan()
 			}
 
 			// Cycle Gcan state and wrap to 0 if needed
-			gcanUpdateState = (gcanUpdateState++) % NUM_GCAN_STATES;
+			gcanUpdateState = (++gcanUpdateState) % NUM_GCAN_STATES;
 
 			service_can_tx(&hcan2);
 		}
 	}
 }
+
+void checkForNewChargerInfo()
+{
+	static uint32_t lastChargerRX = 0;
+	if (newChargerMessage)
+	{
+		lastChargerRX = HAL_GetTick();
+		gBms.chargerConnected = true;
+		updateChargerData(&gBms.chargerData);
+		newChargerMessage = false;
+	}
+	if (gBms.chargerConnected && ((HAL_GetTick() - lastChargerRX) > CHARGER_RX_TIMEOUT_MS))
+	{
+		gBms.chargerConnected = false;
```

Comment:
Is there a need to clear the data in gBms.chargerData? I think we should be fine since most of the alerts check for gBms.chargerConnected which would be false